src/network/networkd-wwan-bus.c

```diff
@@ -0,0 +1,1285 @@
+                sd_bus_message_handler_t callback,
+                void *userdata,
+                Link *link) {
+        _cleanup_(sd_bus_message_unrefp) sd_bus_message *m = NULL;
+        int r;
+
+        assert_return(bus, -EINVAL);
+        assert_return(bus = bus_resolve(bus), -ENOPKG);
+        assert_return(!bus_origin_changed(bus), -ECHILD);
+
+        if (!BUS_IS_OPEN(bus->state))
+                return -ENOTCONN;
+
+        r = sd_bus_message_new_method_call(bus, &m, destination, path, interface, member);
+        if (r < 0)
+                return r;
+
+        r = sd_bus_message_open_container(m, 'a', "{sv}");
+        if (r < 0)
+                return bus_log_create_error(r);
+
+        STRV_FOREACH(prop, link->network->modem_simple_connect_props) {
+                const char *type;
+                _cleanup_free_ char *left = NULL;
+                _cleanup_free_ char *right = NULL;
+                signed int right_i;
+                unsigned int right_u;
+
+                r = split_pair(*prop, "=", &left, &right);
+                type = prop_type_lookup(left);
+                if ((r < 0) || !type) {
+                        log_error("ModemManager: malformed simple connect option: %s, file: %s",
+                                  *prop, link->network->filename);
+                        return -EINVAL;
+                }
+
+                if (streq_ptr(left, "ip-type")) {
+                        MMBearerIpFamily ip_type = prop_iptype_lookup(right);
+
+                        r = sd_bus_message_append(m, "{sv}", left, type, (uint32_t)ip_type);
+                        continue;
+                }
+
+                if (streq_ptr(left, "allowed-auth")) {
+                        MMBearerAllowedAuth auth = prop_auth_lookup(right);
+
+                        r = sd_bus_message_append(m, "{sv}", left, type, (uint32_t)auth);
+                        continue;
+                }
+
+                switch (type[0]) {
+                case SD_BUS_TYPE_BOOLEAN:
+                        r = parse_boolean(right);
+                        if (r < 0)
+                                return -EINVAL;
+                        r = sd_bus_message_append(m, "{sv}", left, type, (bool)r);
+                        break;
+                case SD_BUS_TYPE_INT32:
+                        r = safe_atoi(right, &right_i);
+                        if (r < 0)
+                                return -EINVAL;
+                        r = sd_bus_message_append(m, "{sv}", left, type, (int32_t)right_i);
+                        break;
+                case SD_BUS_TYPE_UINT32:
+                        r = safe_atou(right, &right_u);
+                        if (r < 0)
+                                return -EINVAL;
+                        r = sd_bus_message_append(m, "{sv}", left, type, (uint32_t)right_u);
+                        break;
+                case SD_BUS_TYPE_STRING:
+                        _fallthrough_;
+                default:
```

Comment:
I do not think we gracefully accept `default`, and this should never reached.
Please use `assert_not_reached()`:
```c
default:
        assert_not_reached();
```